components/dashboard/src/start/CreateWorkspace.tsx

```diff
@@ -0,0 +1,102 @@
+import { CreateWorkspaceMode, GitpodService, WorkspaceCreationResult } from "@gitpod/gitpod-protocol";
+import { ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { StartPage, StartPhase } from "../components/StartPage";
+import StartWorkspace from "./StartWorkspace";
+import React, { Suspense } from "react";
+
+const WorkspaceLogs = React.lazy(() => import('./WorkspaceLogs'));
+
+export interface CreateWorkspaceProps {
+  contextUrl: string;
+  gitpodService: GitpodService;
+}
+
+export interface CreateWorkspaceState {
+  result?: WorkspaceCreationResult;
+  error?: CreateWorkspaceError;
+}
+
+export interface CreateWorkspaceError {
+  message?: string;
+  code?: number;
+  data?: any;
+}
+
+export class CreateWorkspace extends React.Component<CreateWorkspaceProps, CreateWorkspaceState> {
```

Comment:
Any particular reason you use a class here?

Reply:
Because it has state -- it needs to call `gitpodService.server.createWorkspace()` exactly once, then wait for the result, then rerender itself when the `result` comes in (or the `error`). I think something like that isn't easily achieved with a function, right?